### PR TITLE
[INLONG-9906][Agent] Add configuration comparison logic and processing of comparison results

### LINKED ISSUE #9906: [Improve][Agent] Add configuration comparison logic and processing of comparison results
### Description

Add configuration comparison logic and processing of comparison results

### InLong Component

InLong Agent

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by justinwwhuang**
[INLONG-9906][Agent] Add configuration comparison logic and processing of comparison results
- Fixes #9906

### Motivation

Add configuration comparison logic and processing of comparison results
### Modifications

Add configuration comparison logic and processing of comparison results

### Verifying this change

*(Please pick either of the following options)*

- [ ] This change is a trivial rework/code cleanup without any test coverage.

- [ ] This change is already covered by existing tests, such as:
  *(please describe tests)*

- [ ] This change added tests and can be verified as follows:

  *(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Documentation

  - Does this pull request introduce a new feature? (yes / no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)
  - If a feature is not applicable for documentation, explain why?
  - If a feature is not documented yet in this PR, please create a follow-up issue for adding the documentation

### CODE REVIEW

File: inlong-agent/agent-installer/src/main/java/org/apache/inlong/agent/installer/ModuleManager.java [hunk 1]
```diff
@@ -216,6 +218,147 @@ private boolean updateModules(List<ModuleConfig> managerModuleList) {
...
+        String ret = ExcuteLinux.exeCmd(module.getCheckCommand());
+        String[] processArray = ret.split("\n");
+        int cnt = 0;
+        for (int i = 0; i < processArray.length; i++) {
+            if (processArray[i].length() > 0) {
+                cnt++;
+            }
+        }
+        LOGGER.info("get module process num {} {}", module.getName(), cnt);
+        if (cnt < module.getProcessesNum()) {
```
Comment (EMsnap) [inline inlong-agent/agent-installer/src/main/java/org/apache/inlong/agent/installer/ModuleManager.java:359]: return cnt >= module.getProcessesNum();

File: inlong-agent/agent-installer/src/main/java/org/apache/inlong/agent/installer/ModuleManager.java [hunk 2]
```diff
@@ -216,9 +218,158 @@ private boolean updateModules(List<ModuleConfig> managerModuleList) {
...
+
+    private boolean saveModuleState(Integer moduleId, ModuleStateEnum state) {
+        ModuleConfig module = currentModules.get(moduleId);
+        if (module == null) {
+            LOGGER.error("should not happen! module {} not found!", moduleId);
+            return false;
+        }
+        LOGGER.info("save module state to {} {}", moduleId, state);
+        module.setState(state);
+        saveToLocalFile(confPath);
```
Comment (aloyszhang) [inline inlong-agent/agent-installer/src/main/java/org/apache/inlong/agent/installer/ModuleManager.java:318]:
```suggestion
        module.setState(state);
        saveToLocalFile(confPath);
        LOGGER.info("save module state to {} {}", moduleId, state);
```

**Review by aloyszhang — APPROVED** [review]
LGTM

**Review by EMsnap — COMMENTED** [review]
LGTM

**Review by EMsnap — APPROVED** [review]

**Pull request merged**